### ISSUE TITLE
Terraform crashes when encountering unexpected version in module blocks

### ISSUE THREAD

**Issue by danc256** [hashicorp/terraform#21414]
### Terraform Version

```
Terraform v0.12.0
+ provider.aws v2.11.0
+ provider.template v2.1.2
```

### Crash Output
https://gist.github.com/danc256/3dece7de90598de89ad62b034f0f8b88

### Expected Behavior
Not crashed almost immediately

### Actual Behavior
It seems every command (init, plan, validate, etc) causes a crash after the 0.12 upgrade

### Steps to Reproduce
Followed the 0.12 upgrade steps steps as provided: https://www.terraform.io/upgrade-guides/0-12.html

### Additional Context
Was running latest v11.x previously without issue on macos. This is a relatively new project (created in the last 15 days).

**Comment by jbardin** [comment]
Hi @danc256,

Thanks for reporting this!
Do you have a configuration you can share that reproduces the issue? I have a feeling this is being triggered by an unexpected `version` field in a module that doesn't support it.

**Comment by danc256** [comment]
Wow that was pretty spot on. There were some modules I grabbed from source because they did some but not all of what I needed and the "version = " tags were still in the modules. Comment those out and crashing is gone. Now 0.12 is picking up on some other slop that I'm looking at. Thanks for the fast reply!

**Comment by apparentlymart** [comment]
This should be fixed by #21542. There were some earlier fixes in similar vein that will also contribute to general robustness in handling version constraints for modules.
